pkg/spi-shared/config/config.go

```diff
@@ -14,22 +14,38 @@
 package config
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"strings"
 
+	"golang.org/x/oauth2"
 	"gopkg.in/yaml.v3"
 )
 
-type ServiceProviderType string
+type ServiceProviderName string
+type ServiceProviderType struct {
+	Name                 ServiceProviderName
+	DefaultOAuthEndpoint oauth2.Endpoint // default oauth endpoint of the service provider
+	DefaultHost          string          // default host of the service provider. ex.: `github.com`
+	DefaultBaseUrl       string          // default base url of the service provider, typically scheme+host. ex: `https://github.com`
+}
 
-const (
-	ServiceProviderTypeGitHub          ServiceProviderType = "GitHub"
-	ServiceProviderTypeQuay            ServiceProviderType = "Quay"
-	ServiceProviderTypeGitLab          ServiceProviderType = "GitLab"
-	ServiceProviderTypeHostCredentials ServiceProviderType = "HostCredentials"
+// all servise provider types we support, including default values
+var SupportedServiceProviderTypes []ServiceProviderType = []ServiceProviderType{
+	ServiceProviderTypeGitHub,
+	ServiceProviderTypeGitLab,
+	ServiceProviderTypeQuay,
```

Comment:
Why is it important to leave out the host creds provider from this list? Maybe it'd be worth mentioning that fact in some kind of comment here...

Reply:
fixed